src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -1088,17 +1088,23 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_AddSrpService(c
         if (strcmp(service.mInstanceName, "") == 0)
         {
             // Assign first empty slot in array for a new service.
-            srpService = srpService ? srpService : &service;
+            srpService = &service;
```

Comment:
I think it was correct before. In case you had an empty slot followed by some service A and you would like to re-add the service A, you would end up trying to add the same service for the second time (and slot).

Reply:
Yes, Damian is right. The assumption is that some empty slot may be followed by some non-empty slot, as we have few services and we can remove any of them from the middle in anytime. So we can't assume that after finding first empty slot there is not anything else in the array.

Reply:
Indeed - So the loop is serving 2 actions:
- lookup for an empty spot
- removal of any duplicate
I'll see if that can be made clearer from the code.

Reply:
> Indeed - So the loop is serving 2 actions:
> 
> * lookup for an empty spot
> * removal of any duplicate

Correct, I agree that it could be documented better, as it might be not so obvious to figure out how it works looking to the code.